tools/gfx/renderer-shared.cpp

```diff
@@ -365,33 +362,25 @@ Result RendererBase::getEntryPointCodeFromShaderCache(
     ComPtr<slang::ISession> session;
     getSlangSession(session.writeRef());
 
-    slang::Digest shaderKeyHash;
-    program->computeDependencyBasedHash(entryPointIndex, targetIndex, &shaderKeyHash);
-
-    StringBuilder shaderKey = hashToString(shaderKeyHash);
+    slang::Digest shaderKey;
+    program->computeDependencyBasedHash(entryPointIndex, targetIndex, &shaderKey);
 
     // Produce a hash using the AST for this program - This is needed to check whether a cache entry is effectively dirty,
     // or to save along with the compiled code into an entry so the entry can be checked if fetched later on.
-    slang::Digest ASTHash;
-    program->computeASTBasedHash(&ASTHash);
-    
+    slang::Digest contentsHash;
+    program->computeASTBasedHash(&contentsHash);
+
     ComPtr<ISlangBlob> codeBlob;
 
-    // Query shaderCacheFileSystem for an entry whose key matches shaderFilename
-    //    - If we find it, then copy the file contents into memory and return in outCode
-    auto result = shaderCacheFileSystem->loadFile(shaderKey.getBuffer(), codeBlob.writeRef());
-    
-    if (SLANG_FAILED(result))
+    // Query the shader cache index for an entry with shaderKeyHash as its key. 
+    auto entry = cacheIndex->findEntry(shaderKey, codeBlob.writeRef());
+    if (!entry)
```

Comment:
Restructure this so you only call getEntryPointCode once from this point.

Reply:
Changed locally, will be in next push